Makefile

```diff
@@ -87,33 +87,35 @@ install-old-locale:
 	@find $(OLD_PREFIX)/share/locale/ -name "guake.mo" -exec rm -f {} \;
 
 install-schemas:
-	install -Dm644 "guake/data/guake.desktop" "$(PREFIX)/share/applications/guake.desktop"
-	install -Dm644 "guake/data/guake-prefs.desktop" "$(PREFIX)/share/applications/guake-prefs.desktop"
-	mkdir -p $(IMAGE_DIR)
-	mkdir -p $(DATA_DIR)
-	install -Dm644 guake/data/pixmaps/*.png "$(IMAGE_DIR)/"
-	install -Dm644 guake/data/pixmaps/*.svg "$(IMAGE_DIR)/"
-	mkdir -p $(SHARE_DIR)
-	install -Dm644  guake/data/*.glade "$(GLADE_DIR)"
-	install -Dm644 "guake/data/org.guake.gschema.xml" "$(SCHEMA_DIR)/"
+	install -Dm644 "guake/data/guake.desktop" "$(INSTALL_ROOT)$(PREFIX)/share/applications/guake.desktop"
+	install -Dm644 "guake/data/guake-prefs.desktop" "$(INSTALL_ROOT)$(PREFIX)/share/applications/guake-prefs.desktop"
+	mkdir -p $(INSTALL_ROOT)$(IMAGE_DIR)
+	install -Dm644 guake/data/pixmaps/*.png "$(INSTALL_ROOT)$(IMAGE_DIR)/"
+	install -Dm644 guake/data/pixmaps/*.svg "$(INSTALL_ROOT)$(IMAGE_DIR)/"
+	install -Dm644 guake/data/pixmaps/guake.png "$(INSTALL_ROOT)$(PREFIX)/share/pixmaps/guake.png"
+	mkdir -p $(INSTALL_ROOT)$(SHARE_DIR)
+	mkdir -p $(INSTALL_ROOT)$(GLADE_DIR)
+	install -Dm644  guake/data/*.glade "$(INSTALL_ROOT)$(GLADE_DIR)"
+	mkdir -p $(INSTALL_ROOT)$(SCHEMA_DIR)
+	install -Dm644 "guake/data/org.guake.gschema.xml" "$(INSTALL_ROOT)$(SCHEMA_DIR)/"
 
 compile-shemas:
-	glib-compile-schemas $(PREFIX)/share/glib-2.0/schemas/
+	glib-compile-schemas $(INSTALL_ROOT)$(PREFIX)/share/glib-2.0/schemas/
```

Comment:
This should use `$(INSTALL_ROOT)$(SCHEMA_DIR)`

Reply:
please use the prefix for that. It should have the root inside it already

Reply:
Yes good catch. But is it needed to build guake? shouldn't  make compile-schemas only be used if you actually install guake in $(PREFIX)? how is this handled in nix, in arch pacamn runs glib-compile-schemas every time you install a new file in /usr/share/glib2.0./...

Reply:
I saw one packager (arch?) that delete this line, don't know why. I need to compile it to debug (dev env) or from /usr/local/... from my ubuntu

Reply:
The concept is not very compatible with Nix’s use of separate immutable prefix for each package. Moreover, Nix also has *profiles*, which are just package directory trees merged into a single tree via `lndir`, simulating FHS. This merger would create a [conficts](https://github.com/NixOS/nixpkgs/issues/1455) between `share/glib-2.0/schemas/gschemas.compiled` files from different packages, so for now, we compile it for each package separately during installation (often handled by makefiles or `meson_post_install` script), and then move the file to `share/gsettings-schemas/${package-name}-${package-version}/glib-2.0/schemas/`.

Reply:
@gsemet pacman (the package manager in arch) runs a hook after the install process which handles the schema compilation. The line is probably removed because during packaging recompiling is not needed.

Reply:
Yes, the compiled file depends not only on the files from the project, but also on everything else that is in the directory. For that reason, it should not be a responsibility of project build scripts (Makefiles) but one of package manager (as pacman does with post-install hooks).

Unfortunately, this falls short when one does not use a package manager (e.g. developer installing the app to `/usr/local/share` for testing), so most project do have `glib-compile-schemas` somewhere in their build scripts. 

One possible “solution” is building the caches (schemas, indexes, etc.) only if `DESTDIR` environment variable [is not set](https://github.com/GNOME/gnome-shell/blob/9f7b1014378a756f761908cc44da0ce07a35ab9d/meson/meson-postinstall.sh) and relying on the fact that most package managers set it.

Reply:
Nice idea! I thought of something more explicit like an opt-out flag (less "magic").

Reply:
I leave that to you, do not hesitate to add this opt-out flag

Reply:
@gsemet isn't that why you added COMPILE_SCHEMA? This works for me on arch.

Reply:
If it is enough for you it’s perfect !